packages/migrations/src/index.ts

```diff
@@ -11,10 +11,23 @@ const sql = sqldb.loadSqlEquiv(__filename);
 export async function init(migrationDir: string, project: string) {
   const lockName = 'migrations';
   logger.verbose(`Waiting for lock ${lockName}`);
-  await namedLocks.doWithLock(lockName, {}, async () => {
-    logger.verbose(`Acquired lock ${lockName}`);
-    await initWithLock(migrationDir, project);
-  });
+  await namedLocks.doWithLock(
+    lockName,
+    {
+      // Migrations *might* take a long time to run, so we'll periodically
+      // renew the lock so that our lock doesn't get killed by the Postgres
+      // idle session timeout.
+      //
+      // That said, we should generally try to keep migrations executing as
+      // quickly as possible. A long-running migration likely means that
+      // Postgres is locking a whole table, which is unacceptable in production.
+      renewPeriodMs: 60_000,
```

Comment:
Rather than having to specify the period here, I'm wondering whether we should just have `autoRenew` be a boolean option. I can't see why different callers would ever want to use different renew periods -- we will just always want the renew period to be longer than the DB idle in transaction timeout, which is a global setting. If we ever change the DB timeout then we'll need to go through and change all the lock renew timeouts to be less than the new DB value.

I thus think we should have a global config option for lock renew timeout, and lock callers can request whether they want auto-renew or not.

Reply:
That's a fair point! I made this configurable in 9af56c7 via `init()` and added a `namedLocksRenewIntervalMs` config option that defaults to a minute.